btrfs-auto-snapshot

```diff
@@ -23,8 +23,9 @@
 #set -o functrace
 #set -o xtrace
 
-# Define our version
-version=2.0.3
+# Define our version. Distributions may replace it with:
+#   sed -i 's/^\(BTRFS_AUTO_SNAPSHOT_VERSION\)=.*/\1=<version>/' <file>
+BTRFS_AUTO_SNAPSHOT_VERSION=${BTRFS_AUTO_SNAPSHOT_VERSION:-unknown}
```

Comment:
The unfortunate part about this is that this number will need to be maintained across releases. Which may slip through, just as it happened with 2.0.4 (the current 2.0.4 still has 2.0.3).

I recon having this "from repo" is not as important as having it correctly in distros, because, when executed from repo, the user needs to run `git pull` themselves, whereas distros tend to upgrade things automatically. Thoughts?

Reply:
Yeah, I missed it in my rush to put out the release. And when I said 'this repo' I really mean GitHub releases. I'm honestly not sure if any distros actually pkg this tool or if everyone gets it from GitHub. I guess we just drop it and see if anyone complains.

Reply:
Bash being bash, it makes things a bit difficult, as there usually is no "compile" step.

E.g. I recently did this with a C++ program: https://github.com/mikesart/inotify-info/pull/35/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R28

I couldn't come up with an ergonomic idea how to munge the release process, update of this number, reasonably automatically.

The best I could come up with was having a `release` shell script that:

1. update the version in the file
2. commit
3. tag
4. update the version to `version-dev` in the file
5. (optional) push the tag

However, I am not convinced having such script is worth it, compared to what you merged now.

 If you have some more thoughts on how to make it ergonomically, let me know.

Reply:
Addendum to the previously linked PR: https://github.com/mikesart/inotify-info/pull/37/files